azure/scope/cluster.go

```diff
@@ -245,7 +252,7 @@ func (s *ClusterScope) LBSpecs() []azure.ResourceSpecGetter {
 			VNetResourceGroup:    s.Vnet().ResourceGroup,
 			SubnetName:           s.ControlPlaneSubnet().Name,
 			FrontendIPConfigs:    s.APIServerLB().FrontendIPs,
-			APIServerPort:        s.APIServerPort(),
```

Comment:
are we still using `APIServerPort()` somewhere else? If not we might want to remove it. If not we might need to rename the function or the property because it could be confusing that we're not setting `APIServerPort` to `APIServerPort()`, what do you think?

Reply:
>are we still using APIServerPort() somewhere else?

Yes, if my [below](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3419#issuecomment-1507872656) understanding is correct, then the `APIServerPort()` is called when configure the frontend port which interacts with azure client [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/controllers/azurecluster_controller.go#L248-L250).

>If not we might need to rename the function or the property because it could be confusing that we're not setting APIServerPort to APIServerPort(), what do you think?

I think it's a good suggestion to rename it. It's so confused with kube-apiserver port.

Reply:
@CecileRobertMichon I rename the `APIServerPort()` func to be `APIServerFrontendPort()`. This func will go to get the frontend port of the API server LB.